src/Service/Modules/Reports/PartsReceivedReportModule.cs

```diff
@@ -13,66 +12,81 @@
+    public class PartsReceivedReportModule : ICarterModule
     {
-        private readonly ITqmsJobRefService tqmsJobRefService;
-
-        private readonly IPartsReceivedReportFacadeService reportFacadeService;
-
-        public PartsReceivedReportModule(
-            ITqmsJobRefService tqmsJobRefService,
-            IPartsReceivedReportFacadeService reportFacadeService)
+        public void AddRoutes(IEndpointRouteBuilder app)
         {
-            this.tqmsJobRefService = tqmsJobRefService;
-            this.reportFacadeService = reportFacadeService;
-            this.Get("/purchasing/tqms-jobrefs", this.GetJobRefs);
-            this.Get("/purchasing/reports/parts-received", this.GetReport);
-            this.Get("/purchasing/reports/parts-received/export", this.GetExport);
+            app.MapGet("/purchasing/tqms-jobrefs", this.GetJobRefs);
+            app.MapGet("/purchasing/reports/parts-received", this.GetReport);
+            app.MapGet("/purchasing/reports/parts-received/export", this.GetExport);
         }
 
-        private async Task GetJobRefs(HttpRequest request, HttpResponse response)
+        private async Task GetJobRefs(
+            HttpRequest request,
+            HttpResponse response,
+            ITqmsJobRefService tqmsJobRefService)
         {
-            var result = this.tqmsJobRefService.GetMostRecentJobRefs(50);
+            var result = tqmsJobRefService.GetMostRecentJobRefs(50);
             await response.Negotiate(result);
         }
 
-        private async Task GetReport(HttpRequest req, HttpResponse res)
+        private async Task GetReport(
+            HttpRequest req,
+            HttpResponse res,
+            int? supplier,
+            string fromDate,
+            string toDate,
+            string jobref,
+            string orderBy,
+            bool includeNegativeValues,
+            IPartsReceivedReportFacadeService reportFacadeService)
         {
             var options = new PartsReceivedReportRequestResource
                               {
-                                  Supplier = req.Query.As<int?>("supplier"),
-                                  FromDate = req.Query.As<string>("fromDate"),
-                                  ToDate = req.Query.As<string>("toDate"),
-                                  Jobref = req.Query.As<string>("jobref"),
-                                  OrderBy = req.Query.As<string>("orderBy"),
-                                  IncludeNegativeValues = req.Query.As<bool>("includeNegativeValues")
+                                  Supplier = supplier,
+                                  FromDate = fromDate,
+                                  ToDate = toDate, 
+                                  Jobref = jobref, 
+                                  OrderBy = orderBy, 
+                                  IncludeNegativeValues = includeNegativeValues
                               };
             if (string.IsNullOrEmpty(options.ToDate))
```

Comment:
I'm using the presence or absence of a required option here to decide whether to return the report or just the view... Could that be causing a problem with the new syntax?

Reply:
Seems not, since I do the same for https://app-sys.linn.co.uk/purchasing/reports/whats-due-in which looks ok